### PR TITLE
Add RBufVector

### PR DESCRIPTION
**Pull request by thestr4ng3r**

### CODE REVIEW

**Comment by thestr4ng3r** [comment]
ping @radare @ret2libc

**Comment by thestr4ng3r** [comment]
Same as the free_all thing would have been.
You can have a single big buffer and let the vector elements point to substrings of it.

File: libr/util/vector.c [hunk 1]
```diff
@@ -297,3 +297,23 @@ static void quick_sort(void **a, size_t n, RPVectorComparator cmp) {
 R_API void r_pvector_sort(RPVector *vec, RPVectorComparator cmp) {
 	quick_sort (vec->v.a, vec->v.len, cmp);
 }
+
```
Comment (ret2libc) [inline libr/util/vector.c]: no need 2 spaces

**Comment by thestr4ng3r** [comment]
Yes, I have a place where I need it. I can show you after I have implemented it.

**Comment by radare** [comment]
why not just add void*user in RVector?

**Comment by thestr4ng3r** [comment]
That would be the same as in https://github.com/radare/radare2/pull/12639, but without the extra free function. So what solution do you want?

**Comment by thestr4ng3r** [comment]
`r_anal_class_get_all()` is the use case and here is a test for it: https://github.com/radare/radare2-regressions/pull/1623/files#diff-dc2281e83ab9f673093d0172b8435e2c

**Comment by thestr4ng3r** [comment]
ping @radare @ret2libc

**Comment by radare** [comment]
r_anal_class_get_all is reachable?

**Comment by thestr4ng3r** [comment]
`r_anal_class_get_all()` is defined as `R_API` and declared in the header, so it is reachable. Or what do you mean?

**Comment by radare** [comment]
It is not called by anyone else

**Comment by thestr4ng3r** [comment]
The test does: https://github.com/radare/radare2-regressions/pull/1623/files#diff-dc2281e83ab9f673093d0172b8435e2c
But I primarily plan to call it from Cutter.

File: libr/anal/class.c [hunk 2]
```diff
@@ -127,6 +127,25 @@ R_API bool r_anal_class_exists(RAnal *anal, const char *name) {
 	return r;
 }
 
+R_API RBufVector/*<const char *>*/ *r_anal_class_get_all(RAnal *anal) {
```
Comment (radare) [inline libr/anal/class.c]: this seems not very efficient, sdb can return a list and you can convert it to vector for faster nth access in cutter. in r2land i dont see the need to have such access need.

**Pull request closed without merging**